src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -315,36 +316,71 @@ export class Selection<T = any> {
     if (this._selectionType !== SelectionType.Multi || !this._items.displayed) {
       return false;
     }
-    const displayedItems: T[] = this._items.displayed;
-    const nbDisplayed = this._items.displayed.length;
+    /** make sure to exclude the locked items from the list */
+    const displayedItems: T[] = this._items.displayed.filter(item => !(this.lockedRefs.indexOf(item) > -1));
+    const nbDisplayed = displayedItems.length;
     if (nbDisplayed < 1) {
       return false;
     }
     const temp: T[] = displayedItems.filter(item => this.current.indexOf(item) > -1);
     return temp.length === displayedItems.length;
   }
 
+  private canItBeLocked() {
+    return [SelectionType.Multi, SelectionType.Single].indexOf(this._selectionType) > -1;
+  }
+
+  /**
+   * Lock and unlock item
+   */
+  public lockItem(item: T, lock: boolean) {
+    if (this.canItBeLocked()) {
+      if (lock === true) {
+        /* Add to lockedRef */
+        this.lockedRefs.push(item);
```

Comment:
If we are storing a ref, we need to first pass it through the trackBy function so we can keep track of the rows. If someone has a dynamic table where they change the list of items, this list of refs will be stale because we'll have new rows.

1) Look above at how trackBy is used to calculate a reference, and also make sure you have a test.
2) When the whole list of items is reset (see above in _allChanges subscription), you'll want to clear the old `lockedRefs` list and let it get rebuilt from the new items, so we don't have a memory leak. The array could continue to grow in side with stale refs otherwise.
3) Anytime you are looking up an item against the `lockedRefs`, you'll have to recompute the ref from the item like in the `isLocked` method below.

Reply:
I still don't see the trackBy here, please review this again.